at_client/lib/src/client/at_client_impl.dart

```diff
@@ -308,19 +325,68 @@ class AtClientImpl implements AtClient {
   @override
   Future<bool> put(AtKey atKey, dynamic value,
       {bool isDedicated = false}) async {
-    // Perform atKey validations.
-    await AtClientValidation.validateAtKey(atKey);
+    AtResponse atResponse = AtResponse();
+    // If the value is neither String nor List<int> throw exception
+    if (value is! String && value is! List<int>) {
```

Comment:
@VJag in put method spec..we allow dynamic value..but in the implementation we check the type to be either string or binary. Is this fine? should the spec be modified to mention only string and binary values are allowed in put?

Reply:
@murali-shris : My reasoning on checking if the value is String or List<int>: Though the older implementation of put method has dynamic, i believe is to support String and List<int> only. The encryption/decryption logic takes input of type only String (For binary data we are encoding to change list<int> to String and vice versa). We do not have parsing logic to convert other data types to String. Further, the data field in AtData object is of type String. Hence restricted the putText to String and putBinary to List<int>. Please let me know your thoughts on this.